### PR TITLE
feat(httplog): add workspace name to the payload 

### PR DESCRIPTION
**Pull request by raoxiaoyan**
### Summary
This PR adds a workspace name to the payload.

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference

FTI-4728

### CODE REVIEW

**Review by hanshuebner — CHANGES_REQUESTED** [review]
Can you please fill out the PR template, removing the boiler plate and ticking the appropriate boxes?  Also, please add an entry to CHANGELOG.md as requested in the template.

File: kong/workspaces/init.lua [hunk 1]
```diff
@@ -48,5 +48,8 @@ function workspaces.get_workspace_id(ctx)
   return (ctx or ngx.ctx).workspace or kong.default_workspace
 end
 
+function workspaces.get_workspace_name()
+    return (workspaces:get_workspace() or { name = "default" }).name
```
Comment (hbagdi) [inline kong/workspaces/init.lua:52]: Please do not create a new table on the fly.

File: spec/03-plugins/03-http-log/01-log_spec.lua [hunk 2]
```diff
@@ -293,6 +293,7 @@ for _, strategy in helpers.each_strategy() do
 
         if #body.entries == 1 then
           assert.same("127.0.0.1", body.entries[1].client_ip)
+          assert.not_nil(body.entries[1].workspace_name)
```
Comment (hbagdi) [inline spec/03-plugins/03-http-log/01-log_spec.lua:296]: Assert the workspace name in one of the test and please make sure to test for default and non-default workspaces.

**Review by hbagdi — CHANGES_REQUESTED** [review]
Please open a documentation PR before you merge this.

File: spec/03-plugins/03-http-log/01-log_spec.lua (hunk 2 above)
Comment (raoxiaoyan) [reply spec/03-plugins/03-http-log/01-log_spec.lua:296]: I see. But the Kong OSS only supports the default workspace. So do we need to test the non-default workspace?

File: kong/workspaces/init.lua (hunk 1 above)
Comment (hanshuebner) [reply kong/workspaces/init.lua]: > Please do not create a new table on the fly.

@hbagdi I looked at this and was not sure if I thought it was clever or ugly, but I was not aware that we have a rule against it.  What is the reason for not doing it this way?

File: spec/03-plugins/03-http-log/01-log_spec.lua (hunk 2 above)
Comment (hbagdi) [reply spec/03-plugins/03-http-log/01-log_spec.lua:296]: Ah. Can you please add a test for this in the other repo (after cherry-picking it)?

File: kong/workspaces/init.lua (hunk 1 above)
Comment (hbagdi) [reply kong/workspaces/init.lua]: If tables are static in nature then creating them on the fly causes unnecessary GC pressure. I'm out of depth with LuaJIT and I'm not sure if the compiler optimizes this specific instance away or not, I prefer being conservative.

File: spec/03-plugins/03-http-log/01-log_spec.lua (hunk 2 above)
Comment (raoxiaoyan) [reply spec/03-plugins/03-http-log/01-log_spec.lua:296]: Sure. I will add a test case in the ee.

**Pull request closed without merging**